src/tests/test_double_compare.cpp

```diff
@@ -0,0 +1,207 @@
+#include <boost/math/special_functions/next.hpp>
+#include <gtest/gtest.h>
+
+#include "types.h"
+
+namespace tests {
+
+class DoubleCompare : public ::testing::Test {
+private:
+    std::unique_ptr<std::byte[]> double_min_ptr;
+    std::unique_ptr<std::byte[]> double_max_ptr;
+    std::unique_ptr<std::byte[]> result_ptr;
+
+protected:
+    void test_arithmetic(model::Double a, model::Double b, model::Double expected,
```

Comment:
I'd rename a and b to left and right, I think it would be more readable

Reply:
Also move this method definition below fields definitions, don't mix them with each other